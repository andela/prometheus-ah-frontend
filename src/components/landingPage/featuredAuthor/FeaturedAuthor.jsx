import React from 'react';
import PropTypes from 'prop-types';
import FeaturedAuthorPost from './FeaturedAuthorPost';

const FeaturedAuthor = (props) => {
  const { author } = props;
  const { featuredAuthor } = author;
  const articles = featuredAuthor.articles.slice(-3);

  const { firstname, lastname, username } = featuredAuthor;

  let authorName = `${firstname} ${lastname}`;

  if (!firstname || !lastname) {
    authorName = username;
  }

  return (
    <div>
      <h2 className="section-title">FEATURED AUTHOR</h2>
      <div className="card shadow-lg border-0 mb-5 featured-author">
        <div className="d-flex ml-3 mt-4">
          <img
            src={featuredAuthor.image
              ? featuredAuthor.image
              : 'https://image.ibb.co/i48Wqf/paceholder.jpg'}
            className="img-fluid featured-author-image rounded-circle"
            alt=""
          />
          <div className="author-detail ml-5 mt-3">
            <p className="author mb-1">
              {authorName}
            </p>
            <button type="button" className="btn btn-follow">Follow</button>
          </div>
        </div>
        <div className="mx-4 mt-5">
          {
            articles.map(post => (
              <FeaturedAuthorPost
                key={post.id}
                title={post.title}
                readTime={post.readingTime}
                slug={post.slug}
              />
            ))
          }
        </div>
      </div>
    </div>
  );
};

FeaturedAuthor.propTypes = {
  author: PropTypes.instanceOf(Object),
};

export default FeaturedAuthor;
