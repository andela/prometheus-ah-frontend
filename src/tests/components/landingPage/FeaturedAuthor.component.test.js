import React from 'react';
import { shallow } from 'enzyme';
import FeaturedAuthor from '../../../components/landingPage/featuredAuthor/FeaturedAuthor';

let wrapper;

describe('FeaturedAuthor component', () => {
  beforeEach(() => {
    const props = {
      author: {
        featuredAuthor: {
          id: 2,
          username: 'faksam',
          firstname: null,
          lastname: null,
          image: null,
          articles: [
            {
              id: 23,
              slug: 'the-most-important-skill-nobody-taught-you',
              title: 'The most important skill nobody taught you',
              readingTime: '1 min read'
            }
          ]
        }
      }
    };

    wrapper = shallow(<FeaturedAuthor {...props} />);
  });

  test('should render a snapshot of the FeaturedAuthor component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
