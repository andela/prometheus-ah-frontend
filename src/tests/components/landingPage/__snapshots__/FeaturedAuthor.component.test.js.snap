// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeaturedAuthor component should render a snapshot of the FeaturedAuthor component 1`] = `
<div
  className="col-12"
>
  <h2
    className="section-title"
  >
    FEATURED AUTHOR
  </h2>
  <div
    className="card shadow-lg border-0 mb-5 featured-author"
  >
    <div
      className="d-flex ml-3 mt-4"
    >
      <img
        alt=""
        className="img-fluid featured-author-image rounded-circle"
        src={null}
      />
      <div
        className="author-detail ml-4 mt-3"
      >
        <p
          className="author mb-1"
        >
          undefined undefined
        </p>
        <button
          className="btn btn-follow"
          type="button"
        >
          Follow
        </button>
      </div>
    </div>
    <div
      className="mx-4 mt-5"
    >
      <FeaturedAuthorPost
        image="https://image.ibb.co/eLVQwL/kids.jpg"
        key="1"
        readTime="1 min read"
        title="My Broken Mother, My Broken Heart"
      />
      <FeaturedAuthorPost
        image="https://image.ibb.co/eLVQwL/kids.jpg"
        key="1"
        readTime="1 min read"
        title="My Broken Mother, My Broken Heart"
      />
    </div>
  </div>
</div>
`;
